nemo_text_processing/text_normalization/normalize.py

```
@@ -12,198 +12,211 @@
+    PYNINI_AVAILABLE = False
 
-def _verbalize(tag: Tag) -> str:
-    """
-    Given tag applies verbalization if possible and returns transduced text.
-    This is context-independent.
-    Args:
-        tag: tag
-    Returns: verbalized text
-    """
-    expand_funcs = VERBALIZERS[tag.kind]
-    res = [f(tag.data) for f in expand_funcs]
-    res = [x for x in res if x]
-    if not res:
-        return tag.text
-    else:
-        return res[0]
-
-
-def normalize(text: str, verbose: bool):
-    """
-    main function. normalizes alphanumerical tokens in given text to its verbalized form:
-    e.g. "12kg -> twelve kilograms"
-    Args:
-        text: string that may include semiotic classes.
-    Returns: verbalized form in string format
-    """
-    tags = find_tags(text)
-    tags = select_tags(tags)
-    output = verbalize(text, tags)
-    if verbose:
-        print(text)
-        print(output)
-        print([str(tag) for tag in tags])
-    return output
 
-
-def normalize_identity(texts: List[str], verbose: bool = False) -> List[str]:
-    """
-    Identity normalizer. Returns input unchanged
-    Args:
-        texts: input string
-    Returns input string
+class Normalizer:
     """
-    return texts
-
+    Normalizer class that converts text from written to spoken form. 
+    Useful for TTS preprocessing. 
 
-def normalize_nemo(texts: List[str], verbose: bool = False) -> List[str]:
-    """
-    Text normalization with NeMo algorithm.
     Args:
-        texts: List of unnormlized strings
-        verbose: if specified prints debugging info
-    Returns list of normalized strings
-    """
-    res = []
-    for input in tqdm(texts):
-        text = normalize(input, verbose=verbose)
-        res.append(text)
-    return res
-
-
-normalizers = {"identity": normalize_identity, "nemo": normalize_nemo}
+        input_case: expected input capitalization
+    """
+
+    def __init__(self, input_case: str):
+        assert input_case in ["lower_cased", "cased"]
```

let's raise an error